### ISSUE TITLE
0.8.0 crashes on decrypt secret

### ISSUE THREAD

**Issue by kraem** [mssun/passforios#308]
Since updating to 0.8.0 (2), I've tried 0.8.0 (3) as well, I can't decrypt any secrets.

Found this due to not being able to AutoFill any passwords.
Later found that as soon as I try to decrypt a secret (click on it in the Pass app) the screen goes black and Pass crashes.

Reverting to 0.7.0 (15) everything is working again.

Phone **iPhone SE**
iOS version **12.4.1 (16G102)**
Pass **0.8.0 (2)** and **0.8.0 (3)**

Happy to provide any more information if needed.

**Comment by yishilin14** [comment]
There are some PGP-related updates since 0.8.0. We need to know how you generate and export your keys.

**Comment by yishilin14** [comment]
Our test cases must have missed some settings.

**Comment by kraem** [comment]
I can't remember right now how I generated them. I think it was with `gpg --full-generate-key`.

The key is a **rsa4096** one.

I've exported them with `gpg --export-secret-keys -a` and `gpg --export -a`. I've then generated QR codes of them with your script to import them.

**Comment by mssun** [comment]
Interesting, in our recent PR, we suppose RSA2048, RSA2048_SUB, ED25519, ED25519_SUB should work with one exception https://github.com/mssun/passforios/pull/293#issuecomment-532965334.

I'm not sure your issue is related with this. Can you give us a detailed instruction to generated this kind of key? Thanks.

@a-feld and @SimplyDanny, there are still cases our pgp library cannot handle, resulting crashes. I guess you guys are more familiar with this kind of issues :)

**Comment by kraem** [comment]
```
$ gpg --full-generate-key 

Please select what kind of key you want:
   (1) RSA and RSA (default)
   (2) DSA and Elgamal
   (3) DSA (sign only)
   (4) RSA (sign only)
Your selection? 1
RSA keys may be between 1024 and 4096 bits long.
What keysize do you want? (2048) 4096
Requested keysize is 4096 bits
Please specify how long the key should be valid.
Key is valid for? (0) 10
Key expires at Sun 29 Sep 2019 20:53:16 CEST
Is this correct? (y/N) y

GnuPG needs to construct a user ID to identify your key.

Real name: testie
Email address: testie@test.com
Comment: 
You selected this USER-ID:
    "testie <testie@test.com>"

Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? O

pub   rsa4096 2019-09-19 [SC] [expires: 2019-09-29]
      5DC2E33372A8D3067ED8EEC426A845371D664CF2
uid                      testie <testie@test.com>
sub   rsa4096 2019-09-19 [E] [expires: 2019-09-29]
```

That's how I created the 4096 RSA key.

**Comment by mssun** [comment]
Thank you. This is very helpful!

**Comment by a-feld** [comment]
Interesting, the key generation and export appears to be pretty typical. I'll try and take a look later this evening to see if I can find a crash.

This may not necessarily be related to the pgp logic though, it could be at a layer above that such as a change in the way things were stored between 0.7 and 0.8 (haven't looked in depth at the diff between these versions)

This could be related to state being interpreted as corrupted in an upgrade for example.

**Comment by SimplyDanny** [comment]
I just generated a new key pair like you did and used it for my test repository. It works without any errors. The only difference is that I could not import the keys via QR codes (the simulator cannot scan images). The iTunes file sharing import mechanism was used instead.

**Comment by kraem** [comment]
@SimplyDanny The import method should not matter, since the key imported is the exact same. Right?

Is the output of `gpg -K` enough to determine the attributes of the key or can I add some other data point?

**Comment by a-feld** [comment]
@kraem Have you tried removing your pgp keys and re-importing them after the upgrade?

**Comment by kraem** [comment]
@a-feld Thanks! It worked when I
1. Erased all password store data from within the app.
2. Removed my GPG key and reimported it using https://github.com/yishilin14/asc-key-to-qr-code-gif (the same RSA 4096 bit key as before)

Guess you can close this if there's no interest investigating why it didn't work without readding the key.

**Comment by a-feld** [comment]
Thanks @kraem for the update, this is very useful for debugging!

@mssun This seems to indicate a problem with the upgrade path from v0.7 -> v0.8 rather than there being an issue with PGP.

**Comment by mssun** [comment]
@a-feld, thanks for the investigation. We need to solve this before rolling out a new version.

**Comment by yishilin14** [comment]
I just fixed a bug (6b00d4911b3f664b6c6e649c5f9e5b978ec05f72). Could be related to this issue.

In 0.8.0 (1~12):
- If `"Settings->Advanced->Encrypt in ASCII-Armored"` is true, decrypting a file not encrypted in ascii-armored format crashes the app.
- If `"Settings->Advanced->Encrypt in ASCII-Armored"` is false, decrypting a file encrypted in ascii-armored format gives "openpgp: invalid data: tag byte does not have MSB set."
 
If your files are encrypted in ascii-armored format, erasing all password store data happens to solve the problem because it resets `"Settings->Advanced->Encrypt in ASCII-Armored"` to false...

**Comment by yishilin14** [comment]
Please try 0.8.0(13).

**Comment by mssun** [comment]
Closed, 0.8.0 (41) has been published in App Store.

Feel free to reopen this or create a new one if there are more related issues.
